bigquery/bigquery.go

```diff
@@ -97,6 +98,17 @@ func NewClient(ctx context.Context, projectID string, opts ...option.ClientOptio
 	return c, nil
 }
 
+// ConfigureStorageReadClient sets up Storage API connection to be used when fetching
+// large datasets from tables, jobs or queries.
+func (c *Client) ConfigureStorageReadClient(ctx context.Context, opts ...option.ClientOption) error {
```

Comment:
Should we just accept an instantiated ReadClient?  You've already exposed NewReadClient().

Additionally, should we block re-configuration?  e.g. once c.rc != nil we emit error on subsequent attempts?  As presented, we'd potentially leak resources from the previous client.

Reply:
Good point, I added a check to see if the client is already initialized here. On passing a `ReadClient` or not, I wasn't sure how much to expose users to that. I was trying to address two groups of users: 

1. Users that just want to enable the Storage API to get faster downloads and don't worry that much about details ( they don't need to know that the `ReadClient` exists for example )
2. Users that know about the Storage API and want to use the ReadClient, created a ReadSession and then read rows from it.